src/lib.rs

```diff
@@ -653,26 +653,27 @@ mod b {
         };
     }
 
-    fn make_this<T: Ord>(v: &mut Vec<T>) -> OrderedCollection<&T> {
-        OrderedCollection::from_slice(v)
+    fn make_this<T: Ord + Copy>(v: &mut Vec<T>) -> OrderedCollection<T> {
+        v.sort_unstable();
+        OrderedCollection::from_sorted_iter(v.into_iter().map(|x| *x))
```

Comment:
I _think_ you can simplify this slightly by using `.iter().copied()`

Reply:
Indeed, thank you! Will file all the followups in a separate PR.